airflow/providers/amazon/aws/sensors/glacier.py

```diff
@@ -0,0 +1,75 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import Any
+
+from airflow.exceptions import AirflowException
+from airflow.providers.amazon.aws.hooks.glacier import GlacierHook
+from airflow.sensors.base_sensor_operator import BaseSensorOperator
+from airflow.utils.decorators import apply_defaults
+
+
+class GlacierJobOperationSensor(BaseSensorOperator):
+    """
+    Glacier sensor for checking job state. This operator runs only in reschedule mode.
+
+    :param vault_name: name of Glacier vault on which job is executed
+    :type vault_name: str
+    :param job_id: the job ID was returned by retrieve_inventory()
+    :type job_id: str
+    """
+
+    template_fields = ["vault_name", "job_id"]
+
+    IN_PROGRESS = "InProgress"
+    SUCCEEDED = "Succeeded"
```

Comment:
WDYT of having a class of type enum?

```python
class JobStatus(Enum):
    IN_PROGRESS = "InProgress"
    SUCCEEDED = "Succeeded"
```

Reply:
This way you are grouping related variables together. :)

Reply:
I was thinking about it. I will add your suggestion. Thanks :)